app/controllers/posts_controller.rb

```diff
@@ -1,10 +1,51 @@
 class PostsController < ApplicationController
   def index
-    @posts = Post.all.order(user_id: :desc)
+    @posts = Post.includes([:user]).where(user_id: params[:user_id])
   end
 
   def show
     @post = Post.find(params[:id])
     @comments = @post.most_recent_five_comments
   end
+
+  def create
+    @post = Post.new(post_params)
+    @post.user = current_user
+    if @post.save
+      redirect_to user_post_path(current_user.id, @post.id)
+    else
+      # @posts = Post.includes([:user]).where(user_id: params[:user_id])
```

Comment:
**Optional**
- Please make sure to remove any unused code for a cleaner codebase